### ISSUE TITLE
Fix false positives for flex shorthand with 2 values in length-zero-no-unit

### ISSUE THREAD

**Issue by Sultanidze** [stylelint/stylelint#4831]
> Clearly describe the bug

There is false negative for `0px` in the flex shorthand rule:  
`flex: 1 0px;`

It is ambiguous situation for two values in flex shorthand property,
according to documentation:
https://developer.mozilla.org/en-US/docs/Web/CSS/flex

#### Two values: flex-grow | flex-basis:
`flex: 1 0px;`

#### Two values: flex-grow | flex-shrink:
`flex: 1 0;`

> Which rule, if any, is the bug related to?

e.g. `length-zero-no-unit`

> What code is needed to reproduce the bug?

```scss
.wrapper {
  flex: 1 0px;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "length-zero-no-unit": true,
  }
}
```

> Which version of stylelint are you using?

`13.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

via `stylelint-webpack-plugin`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No `length-zero-no-unit` error to be flagged.
Possibly it would be good to create another rule to disallow 2 values in flex shorthand property to avoid ambiguous situations, and make intentions evident

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
Value.module.scss
 32:12  ✖  Unexpected unit   length-zero-no-unit
```

**Comment by jeddy3** [comment]
@Sultanidze Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYKBmAbGAPeKAjCgAwAOuA3CgL4gA0IAzgJ4B2ALkrguE0w3AQ2GAJYBzXsAA6bFCmkgATgFccTRQRlz5CkDjbiOACwC0ALxhKIptjZVtRHTSg6qYs+TVl0aQA).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

> Possibly it would be good to create another rule to disallow 2 values in flex shorthand property to avoid ambiguous situations, and make intentions evident

This sounds like a good idea. Can create a separate issue for it, please?

**Comment by Sultanidze** [comment]
@jeddy3 I'm on it.
Will create separate issue for a new rule after fixing current rule.
